src/app/child-dev-project/notes/model/note.ts

```diff
@@ -15,26 +15,38 @@
  *     along with ndb-core.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import { MeetingNoteAttendance } from "../meeting-note-attendance";
 import { DatabaseEntity } from "../../../core/entity/database-entity.decorator";
 import { Entity } from "../../../core/entity/entity";
 import { DatabaseField } from "../../../core/entity/database-field.decorator";
 import { WarningLevel, WarningLevelColor } from "../../warning-level";
 import { InteractionType } from "../note-config-loader/note-config.interface";
+import { EventAttendance } from "../../attendance/model/event-attendance";
+import { AttendanceStatus } from "../../attendance/model/attendance-status";
+import { v4 as uuid } from "uuid";
 
 @DatabaseEntity("Note")
 export class Note extends Entity {
+  static create(date: Date, subject: string = ""): Note {
+    const instance = new Note(uuid());
+    instance.date = date;
+    instance.subject = subject;
+    return instance;
+  }
+
   /** IDs of Child entities linked with this note */
   @DatabaseField() children: string[] = [];
```

Comment:
I think this is redundant with the presence of the `childrenAttendace` attribute.

Reply:
I find the handling of both data structures in this class very confusing and I did not find a good reason to keep the list.

Reply:
There are two reasons:

1. backwards compatibility: existing entities have the children array, so keeping this as the primary property for the linking avoids having to do migration of all existing notes
2. straight-forward (and performant?) use in template - an array can be used directly while a map always has to be transformed to get to the plain list of related ids by calling some filter or method

The idea now is to always have the children array to represent a relationship. If there is additional context about the relation that can be optionally (and completely independently) be added and managed in the childrenAttendance map.